### PR TITLE
Fixed the filer dropdown

### PR DESCRIPTION
**Pull request by aanish27**
#96 Fixed please reveiw @Raza9798

### CODE REVIEW

File: app/Http/Controllers/CompanyController.php [hunk 1]
```diff
@@ -22,51 +22,33 @@ public function draw(Request $request){
...
-            $companys = Company::where('country', 'like', "%" . $searchCountry . "%");
 
-        }else{
+        if(!$request->dropdowns){
             $companys = Company::where('name', 'like', "%" . $search . "%")
                 ->orWhere('country', 'like', "%" . $search . "%")
                 ->orWhere('branch', 'like', "%" . $search . "%")
                 ->orWhere('address', 'like', "%" . $search . "%");
+        }else{
+            $searchCountry = $request->dropdowns['country'] ?? null;
```
Comment (Raza9798) [inline app/Http/Controllers/CompanyController.php:32]: use equal conditions for filter dropdown

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CompanyController.php (hunk 1 above)
Comment (aanish27) [reply app/Http/Controllers/CompanyController.php:32]: Sure did @Raza9798 please review again

Comment (aanish27) [reply app/Http/Controllers/CompanyController.php:32]: Which is better approach?
image

image

Comment (Raza9798) [reply app/Http/Controllers/CompanyController.php:32]: this condition won't work if we choose multiple items in the filter please fix it

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CompanyController.php (hunk 1 above)
Comment (aanish27) [reply app/Http/Controllers/CompanyController.php:32]: Fixed please review again @Raza9798

**Comment by aanish27** [comment]
>this condition won't work if we choose multiple items in the filter please fix it

fixed @Raza9798

File: resources/views/dashboard/company/index.blade.php [hunk 2]
```diff
@@ -109,7 +109,7 @@
             </div>
 
             <div id="table-filters" class="row row-cols-auto gap-2 mt-4 ms-0">
-                <select id="filter-country" class="form-control-sm col-1" aria-label="Default select example">
+                <select id="filter-country" class="form-control-sm col-1" aria-label="Default select example" multiple>
```
[the remark this answers is not in the document]
Comment (aanish27) [reply resources/views/dashboard/company/index.blade.php:112]: can you share the configurations? still couldn't find a way to make it work

[the remark this answers is not in the document]
Comment (Raza9798) [reply resources/views/dashboard/company/index.blade.php:112]: https://developer.snapappointments.com/bootstrap-select/ @aanish27

[the remark this answers is not in the document]
Comment (aanish27) [reply resources/views/dashboard/company/index.blade.php:112]: Tried everything yet couldn't make it work @aanish27

[the remark this answers is not in the document]
Comment (Raza9798) [reply resources/views/dashboard/company/index.blade.php:112]: https://select2.org/ use this package to implement the filter

[the remark this answers is not in the document]
Comment (aanish27) [reply resources/views/dashboard/company/index.blade.php]: > https://select2.org/ use this package to implement the filter

Implemented please review again @Raza9798

File: resources/views/components/dropdown-filter.blade.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<div class="filter p-0" >
+    <label for="{{ $id }}" class="bi bi-filter"> {{ $name }}
```
Comment (Raza9798) [inline resources/views/components/dropdown-filter.blade.php:2]: label and select tags should be separated inside the div. select tag cannot be wrapped by label tag

File: resources/views/dashboard/employee/index.blade.php [hunk 4]
```diff
@@ -316,6 +292,23 @@ class: 'bi bi-plus-lg btn btn-outline-primary rounded-2 px-3 py-0' ,
                     ],
                 });
 
+                //Dropdown filter
+                $('.select2-filter').on('change', function () {
+                    const select = $(this)
+                    const ul = $(this).siblings('span.select2').find('ul')
+                    const count = $(this).select2('data').length
+                    if($(this).val().includes("clear-btn")) {
```
Comment (Raza9798) [inline resources/views/dashboard/employee/index.blade.php:300]: remove this if block and use inbuild clear option

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: resources/views/components/dropdown-filter.blade.php (hunk 3 above)
Comment (aanish27) [reply resources/views/components/dropdown-filter.blade.php]: Yes got it...i refactored please review again

File: resources/views/dashboard/employee/index.blade.php (hunk 4 above)
Comment (aanish27) [reply resources/views/dashboard/employee/index.blade.php]: implemented as mentioned please review again @Raza9798 #96

**Pull request merged**